inst/BS5/assets/pkgdown.scss

```diff
@@ -495,3 +495,68 @@ mark {
 .algolia-autocomplete .aa-dropdown-menu .aa-suggestion.aa-cursor {
   background-color: mix($primary, $body-bg, 10%);
 }
+
+/* ---- Dark Mode ---- */
+[data-bs-theme="dark"] {
+  .navbar.bg-info {
+    background-color: #333f48 !important;
+  }
+
+  ::selection {
+    background: #344c5a;
+  }
+
+  input[type="search"] {
+    border-color: var(--bs-border-color);
+  }
+
+  pre {
+    background-color: #272b30;
+  }
+
+  // low-tech image softening in dark mode
+  img {
+    opacity: 0.66;
+    transition: opacity 250ms ease-in-out;
+  }
+
+  img:hover,
+  img:focus,
+  img:active {
+    opacity: 1;
+  }
+
+  // github-dark.css //
```

Comment:
This should probably be supported in a more user-friendly way, but the concept is that you can inline the highlight styles in https://github.com/r-lib/pkgdown/tree/c9206802f2888992de92aa41f517ba7812f05331/inst/highlight-styles in a `[data-bs-theme="dark"] { ... }` context.

Reply:
Currently the light syntax highlight theme is set by `template.theme`, so I propose we add `template.theme-dark` and default to `arrow-dark`. Then we need to somehow wrap that in `[data-bs-theme="dark"] {}` in `bs_theme_rules()`. Worse case we can just dump in a temporary file scoped to the parent frame, but maybe there's a smarter way.